I need to also fix the compilation with clang + samples:

```
[ 55%] Building CXX object samples/CMakeFiles/exiv2json.dir/exiv2json.cpp.o
In file included from /src/exiv2/samples/exiv2json.cpp:24:
/src/exiv2/samples/Jzon.h:211:32: error: 'iterator<std::input_iterator_tag, std::pair<std::string, Jzon::Node &>>' is deprecated [-Werror,-Wdeprecated-declarations]
                class iterator : public std::iterator<std::input_iterator_tag, NamedNode>
                                             ^
/usr/local/bin/../include/c++/v1/__iterator/iterator.h:24:29: note: 'iterator<std::input_iterator_tag, std::pair<std::string, Jzon::Node &>>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 iterator
```

--- a/.github/workflows/on_PR_linux_fuzz.yml
+++ b/.github/workflows/on_PR_linux_fuzz.yml
@@ -16,12 +16,24 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: install dependencies
-      run: sudo ./ci/install_dependencies.sh
+      run: |
+        sudo ./ci/install_dependencies.sh
+        sudo apt-get install ninja-build
+
     - name: build and compile
       run: |
-        mkdir build && cd build
-        cmake -DEXIV2_ENABLE_PNG=ON -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_ENABLE_BMFF=ON -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DCMAKE_CXX_COMPILER=$(which clang++) -DEXIV2_BUILD_FUZZ_TESTS=ON -DEXIV2_TEAM_USE_SANITIZERS=ON ..
-        make -j $(nproc)
+        mkdir build && cd build && \
+        cmake -GNinja -DEXIV2_ENABLE_PNG=ON \
+          -DEXIV2_BUILD_SAMPLES=ON \
+          -DEXIV2_ENABLE_WEBREADY=ON \
+          -DEXIV2_ENABLE_CURL=ON \
+          -DEXIV2_ENABLE_BMFF=ON \
+          -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON \
+          -DCMAKE_CXX_COMPILER=$(which clang++) \
+          -DEXIV2_BUILD_FUZZ_TESTS=ON \
+          -DEXIV2_TEAM_USE_SANITIZERS=ON \
+          .. && \
+        cmake --build .
 
     - name: Fuzz
       run: |

--- a/.github/workflows/on_PR_linux_matrix.yml
+++ b/.github/workflows/on_PR_linux_matrix.yml
@@ -36,8 +36,19 @@ jobs:
 
       - name: Build
         run: |
-          cd build
-          cmake -GNinja -DCMAKE_BUILD_TYPE=${{matrix.build_type}} -DBUILD_SHARED_LIBS=${{matrix.shared_libraries}} -DEXIV2_ENABLE_PNG=ON -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_BUILD_UNIT_TESTS=ON -DEXIV2_ENABLE_BMFF=ON -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DCMAKE_INSTALL_PREFIX=install ..
+          cd build && \
+          cmake -GNinja \
+            -DCMAKE_BUILD_TYPE=${{matrix.build_type}} \
+            -DBUILD_SHARED_LIBS=${{matrix.shared_libraries}} \
+            -DEXIV2_BUILD_SAMPLES=ON \
+            -DEXIV2_ENABLE_PNG=ON \
+            -DEXIV2_ENABLE_WEBREADY=ON \
+            -DEXIV2_ENABLE_CURL=ON \
+            -DEXIV2_BUILD_UNIT_TESTS=ON \
+            -DEXIV2_ENABLE_BMFF=ON \
+            -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON \
+            -DCMAKE_INSTALL_PREFIX=install \
+            .. && \
           cmake --build .
 
       - name: Install

--- a/.github/workflows/on_PR_linux_special_buils.yml
+++ b/.github/workflows/on_PR_linux_special_buils.yml
@@ -114,8 +114,21 @@ jobs:
 
       - name: Build
         run: |
-          cd build
-          cmake -GNinja -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DEXIV2_ENABLE_PNG=ON -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_BUILD_UNIT_TESTS=ON -DEXIV2_ENABLE_BMFF=ON -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DBUILD_WITH_COVERAGE=OFF -DEXIV2_TEAM_USE_SANITIZERS=ON -DCMAKE_INSTALL_PREFIX=install ..
+          cd build && \
+          cmake -GNinja \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DBUILD_SHARED_LIBS=ON \
+            -DEXIV2_BUILD_SAMPLES=ON \
+            -DEXIV2_ENABLE_PNG=ON \
+            -DEXIV2_ENABLE_WEBREADY=ON \
+            -DEXIV2_ENABLE_CURL=ON \
+            -DEXIV2_BUILD_UNIT_TESTS=ON \
+            -DEXIV2_ENABLE_BMFF=ON \
+            -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON \
+            -DBUILD_WITH_COVERAGE=OFF \
+            -DEXIV2_TEAM_USE_SANITIZERS=ON \
+            -DCMAKE_INSTALL_PREFIX=install \
+            .. && \
           cmake --build .
 
       - name: Tests
@@ -150,8 +163,21 @@ jobs:
 
       - name: Build
         run: |
-          cd build
-          cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DEXIV2_ENABLE_PNG=ON -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_BUILD_UNIT_TESTS=ON -DEXIV2_ENABLE_BMFF=ON -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DBUILD_WITH_COVERAGE=ON -DEXIV2_BUILD_DOC=ON -DEXIV2_ENABLE_NLS=ON -DCMAKE_CXX_FLAGS="-DEXIV2_DEBUG_MESSAGES" ..
+          cd build && \
+          cmake -DCMAKE_BUILD_TYPE=Release \
+            -DBUILD_SHARED_LIBS=ON \
+            -DEXIV2_BUILD_SAMPLES=ON \
+            -DEXIV2_ENABLE_PNG=ON \
+            -DEXIV2_ENABLE_WEBREADY=ON \
+            -DEXIV2_ENABLE_CURL=ON \
+            -DEXIV2_BUILD_UNIT_TESTS=ON \
+            -DEXIV2_ENABLE_BMFF=ON \
+            -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON \
+            -DBUILD_WITH_COVERAGE=ON \
+            -DEXIV2_BUILD_DOC=ON \
+            -DEXIV2_ENABLE_NLS=ON \
+            -DCMAKE_CXX_FLAGS="-DEXIV2_DEBUG_MESSAGES" \
+            .. && \
           make -j
 
       - name: Generate documentation

--- a/.github/workflows/on_PR_mac_matrix.yml
+++ b/.github/workflows/on_PR_mac_matrix.yml
@@ -30,8 +30,20 @@ jobs:
 
       - name: Build
         run: |
-          mkdir build && cd build
-          cmake -GNinja -DCMAKE_BUILD_TYPE=${{matrix.build_type}} -DBUILD_SHARED_LIBS=${{matrix.shared_libraries}} -DEXIV2_ENABLE_PNG=ON -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_BUILD_UNIT_TESTS=ON -DEXIV2_ENABLE_BMFF=ON -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DCMAKE_INSTALL_PREFIX=install -DCMAKE_CXX_FLAGS="-Wno-deprecated-declarations" ..
+          mkdir build && cd build && \
+          cmake -GNinja \
+            -DCMAKE_BUILD_TYPE=${{matrix.build_type}} \
+            -DBUILD_SHARED_LIBS=${{matrix.shared_libraries}} \
+            -DEXIV2_BUILD_SAMPLES=ON \
+            -DEXIV2_ENABLE_PNG=ON \
+            -DEXIV2_ENABLE_WEBREADY=ON \
+            -DEXIV2_ENABLE_CURL=ON \
+            -DEXIV2_BUILD_UNIT_TESTS=ON \
+            -DEXIV2_ENABLE_BMFF=ON \
+            -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON \
+            -DCMAKE_INSTALL_PREFIX=install \
+            -DCMAKE_CXX_FLAGS="-Wno-deprecated-declarations" \
+            .. && \
           cmake --build .
 
       - name: Install

--- a/.github/workflows/on_PR_windows_matrix.yml
+++ b/.github/workflows/on_PR_windows_matrix.yml
@@ -134,6 +134,7 @@ jobs:
             -DCMAKE_CXX_FLAGS=-Wno-deprecated \
             -DCMAKE_BUILD_TYPE=${{matrix.build_type}} \
             -DBUILD_SHARED_LIBS=${{matrix.shared_libraries}} \
+            -DEXIV2_BUILD_SAMPLES=ON \
             -DEXIV2_ENABLE_NLS=ON \
             -DEXIV2_ENABLE_WIN_UNICODE=ON \
             -DEXIV2_ENABLE_WEBREADY=ON \

--- a/.github/workflows/on_push_BasicWinLinMac.yml
+++ b/.github/workflows/on_push_BasicWinLinMac.yml
@@ -44,9 +44,20 @@ jobs:
 
       - name: Build
         run: |
-          cd build
-          cmake -GNinja -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DEXIV2_ENABLE_NLS=OFF -DEXIV2_ENABLE_PNG=ON -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_BMFF=ON -DEXIV2_BUILD_UNIT_TESTS=ON -DEXIV2_ENABLE_WIN_UNICODE=OFF -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DCMAKE_INSTALL_PREFIX=install ..
-          cmake --build .
+          cmake -GNinja `
+            -DCMAKE_BUILD_TYPE=Release `
+            -DBUILD_SHARED_LIBS=ON `
+            -DEXIV2_BUILD_SAMPLES=ON `
+            -DEXIV2_ENABLE_NLS=OFF `
+            -DEXIV2_ENABLE_PNG=ON `
+            -DEXIV2_ENABLE_WEBREADY=ON `
+            -DEXIV2_ENABLE_BMFF=ON `
+            -DEXIV2_BUILD_UNIT_TESTS=ON `
+            -DEXIV2_ENABLE_WIN_UNICODE=OFF `
+            -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON `
+            -DCMAKE_INSTALL_PREFIX=install .. `
+            -S . -B build && `
+          cmake --build build
 
 
       - name: Test
@@ -76,8 +87,19 @@ jobs:
 
       - name: build and compile
         run: |
-          cd build
-          cmake -GNinja -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_BUILD_UNIT_TESTS=ON -DEXIV2_ENABLE_BMFF=ON -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DCMAKE_INSTALL_PREFIX=install -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON ..
+          cd build && \
+          cmake -GNinja \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DBUILD_SHARED_LIBS=ON \
+            -DEXIV2_BUILD_SAMPLES=ON \
+            -DEXIV2_ENABLE_PNG=ON \
+            -DEXIV2_ENABLE_WEBREADY=ON \
+            -DEXIV2_ENABLE_CURL=ON \
+            -DEXIV2_BUILD_UNIT_TESTS=ON \
+            -DEXIV2_ENABLE_BMFF=ON \
+            -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON \
+            -DCMAKE_INSTALL_PREFIX=install \
+            .. && \
           cmake --build .
 
       - name: Test
@@ -106,8 +128,20 @@ jobs:
 
       - name: build and compile
         run: |
-          mkdir build && cd build
-          cmake -GNinja -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_BUILD_UNIT_TESTS=ON -DEXIV2_ENABLE_BMFF=ON -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DCMAKE_INSTALL_PREFIX=install -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DCMAKE_CXX_FLAGS="-Wno-deprecated-declarations" ..
+          mkdir build && cd build && \
+          cmake -GNinja \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DBUILD_SHARED_LIBS=ON \
+            -DEXIV2_BUILD_SAMPLES=ON \
+            -DEXIV2_ENABLE_PNG=ON \
+            -DEXIV2_ENABLE_WEBREADY=ON \
+            -DEXIV2_ENABLE_CURL=ON \
+            -DEXIV2_BUILD_UNIT_TESTS=ON \
+            -DEXIV2_ENABLE_BMFF=ON \
+            -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON \
+            -DCMAKE_INSTALL_PREFIX=install \
+            -DCMAKE_CXX_FLAGS="-Wno-deprecated-declarations" \
+            .. && \
           cmake --build .
 
       - name: Test

--- a/.github/workflows/on_push_ExtraJobsForMain.yml
+++ b/.github/workflows/on_push_ExtraJobsForMain.yml
@@ -33,8 +33,19 @@ jobs:
 
       - name: Build
         run: |
-          cd build
-          cmake -DCMAKE_BUILD_TYPE=Debug -DBUILD_SHARED_LIBS=ON -DEXIV2_ENABLE_PNG=ON -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_BUILD_UNIT_TESTS=ON -DEXIV2_ENABLE_BMFF=ON -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DBUILD_WITH_COVERAGE=ON -DCMAKE_INSTALL_PREFIX=install ..
+          cd build && \
+          cmake -DCMAKE_BUILD_TYPE=Debug \
+            -DBUILD_SHARED_LIBS=ON \
+            -DEXIV2_BUILD_SAMPLES=ON \
+            -DEXIV2_ENABLE_PNG=ON \
+            -DEXIV2_ENABLE_WEBREADY=ON \
+            -DEXIV2_ENABLE_CURL=ON \
+            -DEXIV2_BUILD_UNIT_TESTS=ON \
+            -DEXIV2_ENABLE_BMFF=ON \
+            -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON \
+            -DBUILD_WITH_COVERAGE=ON \
+            -DCMAKE_INSTALL_PREFIX=install \
+            .. && \
           make -j
 
       - name: Tests + Upload coverage

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,7 +32,7 @@ option( EXIV2_ENABLE_WEBREADY         "Build webready support into library"     
 option( EXIV2_ENABLE_CURL             "USE Libcurl for HttpIo (WEBREADY)"                     OFF )
 option( EXIV2_ENABLE_BMFF             "Build with BMFF support"                               ON  )
 
-option( EXIV2_BUILD_SAMPLES           "Build sample applications"                             ON  )
+option( EXIV2_BUILD_SAMPLES           "Build sample applications"                             OFF )
 option( EXIV2_BUILD_EXIV2_COMMAND     "Build exiv2 command-line executable"                   ON  )
 option( EXIV2_BUILD_UNIT_TESTS        "Build unit tests"                                      OFF )
 option( EXIV2_BUILD_FUZZ_TESTS        "Build fuzz tests (libFuzzer)"                          OFF )

--- a/include/exiv2/jpgimage.hpp
+++ b/include/exiv2/jpgimage.hpp
@@ -49,11 +49,11 @@ namespace Exiv2 {
      */
     struct EXIV2API Photoshop {
         // Todo: Public for now
-        inline static const char* const ps3Id_ = "Photoshop 3.0\0"; //!< %Photoshop marker
-        inline static const std::array<const char*, 4> irbId_{"8BIM", "AgHg", "DCSR", "PHUT"};  //!< %Photoshop IRB markers
-        inline static const char* const bimId_ = "8BIM";  //!< %Photoshop IRB marker (deprecated)
-        inline static const uint16_t iptc_ = 0x0404;      //!< %Photoshop IPTC marker
-        inline static const uint16_t preview_ = 0x040c;   //!< %Photoshop preview marker
+        static constexpr std::array<const char*, 4> irbId_{"8BIM", "AgHg", "DCSR", "PHUT"}; //!< %Photoshop IRB markers
+        inline static const char* ps3Id_ = "Photoshop 3.0\0"; //!< %Photoshop marker
+        inline static const char* bimId_ = "8BIM";            //!< %Photoshop IRB marker (deprecated)
+        inline static const uint16_t iptc_ = 0x0404;          //!< %Photoshop IPTC marker
+        inline static const uint16_t preview_ = 0x040c;       //!< %Photoshop preview marker
 
         /*!
           @brief Checks an IRB

--- a/samples/Jzon.cpp
+++ b/samples/Jzon.cpp
@@ -25,7 +25,7 @@ THE SOFTWARE.
 #define JzonAPI __declspec(dllexport)
 #endif
 
-#ifdef WIN32
+#ifdef _MSC_VER
 #define _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS
 #endif
 

